### DISCUSSION TITLE
metrics loss when use pushgateway with python

### DISCUSSION THREAD

**Discussion by wonderisland** [prometheus/pushgateway#558] [Q&A]
http://192.168.41.156:9091/metrics/job/gpt2
# HELP hectorgao_seconds_stat op timecost\n# TYPE hectorgao_seconds_stat histogram\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.005",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.01",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.025",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.05",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.075",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.1",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.25",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.5",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="0.75",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="1.0",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="2.5",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="5.0",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="7.5",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="10.0",name="test_op",op_name="test_op",rank="0",service="gpt2"} 0.0\nhectorgao_seconds_stat_bucket{env="stage",instance="localhost",job="gpt2",le="+Inf",name="test_op",op_name="test_op",rank="0",service="gpt2"} 1.0\nhectorgao_seconds_stat_count{env="stage",instance="localhost",job="gpt2",name="test_op",op_name="test_op",rank="0",service="gpt2"} 1.0\nhectorgao_seconds_stat_sum{env="stage",instance="localhost",job="gpt2",name="test_op",op_name="test_op",rank="0",service="gpt2"} 100.0\n# HELP hectorgao_seconds_stat_created op timecost\n# TYPE hectorgao_seconds_stat_created gauge\nhectorgao_seconds_stat_created{env="stage",instance="localhost",job="gpt2",name="test_op",op_name="test_op",rank="0",service="gpt2"} 1.6870270143945758e+09\n

**Comment by wonderisland** [comment]
could anyone give me some idea and direction, so confusing.

**Comment by beorn7** [comment]
What is your question?
